### PR TITLE
Techniques intro

### PR DESCRIPTION
**Pull request by nathanshammah**
## Description

---
Address #1541 by populating with workflow images and a brief introduction the various QEM techniques in the User Guide. 

One question (@andreamari @Misty-W @natestemen @nickdgardner etc.): Would it make sense, if these become sort of a overview page to collect info about the technique, to list under the summary for the User Guide's sections (1-5), also the relevant Examples, or it would be confusing, as this is the User Guide and it's a separate thing from the Examples? I think it could be nice and an additional way to explore the documentation.

### License

- [x] I license this contribution under the terms of the GNU GPL, version 3 and grant Unitary Fund the right to provide additional permissions as described in section 7 of the GNU GPL, version 3.

Before opening the PR, please ensure you have completed the following where appropriate.
- [x] I added unit tests for new code.
- [x] I used [type hints](https://www.python.org/dev/peps/pep-0484/) in function signatures.
- [x] I used [Google-style](https://google.github.io/styleguide/pyguide.html#383-functions-and-methods) docstrings for functions.
- [x] I [updated the documentation](../blob/master/docs/CONTRIBUTING_DOCS.md) where relevant.
- [x] Added myself / the copyright holder to the AUTHORS file

### CODE REVIEW

**Comment by natestemen** [comment]
> Would it make sense, if these become sort of a overview page to collect info about the technique

Yeah, I think that could work really nicely. No need, IMO, to have a clear line drawn between the examples and user guide.

**Comment by andreamari** [comment]
> One question (@andreamari @Misty-W @natestemen @nickdgardner etc.): Would it make sense, if these become sort of a overview page to collect info about the technique, to list under the summary for the User Guide's sections (1-5), also the relevant Examples, or it would be confusing, as this is the User Guide and it's a separate thing from the Examples? I think it could be nice and an additional way to explore the documentation.

Sounds good! 
It can be hard to keep up-to-date those lists of examples. Maybe we can select one or two examples for each technique (if existing) and say something like:
```
Here are some examples:
- ex 1
- ex 2
You can find more in the Examples (link) section of the documentation.
```

But I am also fine with listing all the examples for each technique. And if it goes out of date it's not a big issue.

File: docs/source/guide/ddd.myst [hunk 1]
```diff
@@ -1,5 +1,25 @@
 # Digital Dynamical Decoupling
 
+Digital Dynamical Decoupling (DDD) is an error mitigation technique in which
+sequences of gates are applied to slack windows, i.e. single-qubit idle
+windows, in a quantum circuit. Such sequences of gates can reduce the coupling
+between the qubits and the environment, mitigating the effects of noise.
+For more discussion of the theory of DDD, see the section [What is the theory
+behind DDD?](ddd-5-theory.myst). Below you can find sections of the
+documentation that address also the following questions:
```
Comment (andreamari) [inline docs/source/guide/ddd.myst:9]:
```suggestion
behind DDD?](ddd-5-theory.myst).
```

Comment (andreamari) [reply docs/source/guide/ddd.myst:9]: The same sentence is already written a few lines later.

File: docs/source/guide/pec.myst [hunk 2]
```diff
@@ -1,6 +1,18 @@
 (guide/pec/pec)=
 # Probabilistic error cancellation
 
+Probabilistic error cancellation (PEC) is an error mitigation technique for improving the estimation of expectation values (see [What is the theory behind PEC?](pec-5-theory.myst)).
```
Comment (andreamari) [inline docs/source/guide/pec.myst:4]:
```suggestion
Probabilistic error cancellation (PEC) is an error mitigation technique for improving the estimation of expectation values. 
In PEC, ideal operations are represented as linear combinations of noisy operations and unbiased estimates of expectation values are obtained by Monte Carlo averaging over different noisy circuits (see [What is the theory behind PEC?](pec-5-theory.myst)).
```

**Review by andreamari — APPROVED** [review]
LGTM, very nice!  
I also suggested a single-sentence description for PEC. Feel free to change it.

@nickdgardner, if you decide to add QEM techniques to the glossary, you can link to the summary pages introduced in this PR.

**Review by natestemen — COMMENTED** [review]
Thanks Nathan, I think this is a great improvement!

I've left a few comments, but shouldn't require anything major.

**Comment by nickdgardner** [comment]
Yes I think listing links to relevant Examples in this part of the User Guide is a nice idea. (And in general the changes look great to me!)

> @nickdgardner, if you decide to add QEM techniques to the glossary, you can link to the summary pages introduced in this PR.

@andreamari Will do!

**Comment by nathanshammah** [comment]
Thanks @natestemen, I implemented the suggested changes. Feel free to merge if it looks fine.

**Review by natestemen — APPROVED** [review]
🚢

**Comment by Misty-W** [comment]
Fixes #1541

**Pull request merged**